src/rossum-sa-extension/scripts/list-selector-improvements.js

```diff
@@ -1,37 +1,66 @@
 // @flow
 
-const listSelectorImprovements = document.createElement('style');
-
-// TODO: apply only on /settings/extensions/logs (?)
-listSelectorImprovements.textContent = `
-  div:has(> ul.MuiMenu-list) {
-    min-width: 400px !important;
-  }`;
-
-document.head?.appendChild(listSelectorImprovements);
+function findAddedDOMElement(mutation, querySelector) {
+  if (mutation.type === 'childList' && mutation.addedNodes.length > 0) {
+    // Assume there is always only one node in the list
+    for (const node of mutation.addedNodes) {
+      if (node instanceof HTMLElement) {
+        // Query the parent in case we searched for the node itself (querySelector works only for descendants)
+        return node.parentNode.querySelector(querySelector);
+      }
+    }
+  }
+  return undefined;
+}
 
 const listSelectorObserver = new MutationObserver((mutations) => {
   for (const mutation of mutations) {
-    if (mutation.type === 'childList' && mutation.addedNodes.length > 0) {
-      for (const node of mutation.addedNodes) {
-        if (node instanceof HTMLElement) {
-          const element = node.querySelector('ul.MuiMenu-list');
-          if (element) {
-            // TODO: remove the extension ID and append it to the end of the item name
-            Array.from(element.getElementsByTagName('li'))
-              .sort((a, b) => a.textContent.localeCompare(b.textContent))
-              .forEach((li) => element.appendChild(li));
-          }
-        }
-      }
+    const element = findAddedDOMElement(mutation, 'ul.MuiMenu-list');
+    if (element) {
+      // TODO: remove the extension ID and append it to the end of the item name
+      Array.from(element.getElementsByTagName('li'))
+        .sort((a, b) => a.textContent.localeCompare(b.textContent))
+        .forEach((li) => element.appendChild(li));
+    }
+  }
+});
+
+const extensionTableObserver = new MutationObserver((mutations) => {
+  for (const mutation of mutations) {
+    const tbody = findAddedDOMElement(mutation, 'tbody');
+    if (tbody && (tbody?.children ?? []).length > 0) {
+      const sortedChildren = Array.from(tbody.children).sort((a, b) =>
+        a.textContent.localeCompare(b.textContent),
+      );
+
+      tbody.replaceChildren(...sortedChildren);
     }
   }
 });
 
 const body = document.body;
 if (body) {
-  listSelectorObserver.observe(body, {
-    childList: true,
-    subtree: true,
-  });
+  if (window.location.pathname.endsWith('/extensions/logs')) {
```

Comment:
I am afraid this will only work when you refresh the page, and you are already on this URL. However, if you open the page on another URL and transition to this path, it will not run. This would need to be done differently (not sure how, quite frankly). Am I missing something important? 🤔